File: Testsuite/Stress/src/fibfp.c
/* FIBFP -- Computes fib(25) using floating point */

#include <stdio.h>

#define FLOAT double

static FLOAT fib (n)
FLOAT n;
{
  if (n < 2.)
    return n;
  else
    return fib (n-1.) + fib (n-2.);
}

int main (argc, argv)
int argc;
char *argv[];
{
  int i;
  FLOAT result;

  for (i=0; i<50; i++)
    result = fib (25.);

  if (result != 75025.)
    printf ("*** wrong result ***\n");

  return 0;
}

File: Testsuite/Benchmarking/CrossPlatform/src/fibfp.java
/* FIBFP -- Computes fib(25) using floating point */

class fibfp {

    static double fib (double n)
    {
      if (n < 2.0)
        return n;
      else
        return fib (n-1.0) + fib (n-2.0);
    }

    public static void main (String args[])
    {
      int i;
      double result = 0.0;

      for (i=0; i<50; i++)
        result = fib (25.0);

      if (result != 75025.0)
        System.out.println ("*** wrong result ***");
    }

}
